test/testdata/rewriter/flatten_nested_sclass.rb

```diff
@@ -0,0 +1,33 @@
+# typed: true
+
+class A
+  def self.foo;
+    class << self
+      def bar; end
+    end
+  end
+end
+
+A.foo
+
+class B
+  def foo;
+    class << self
+      def bar; end
+    end
+  end
+end
+
+B.new.foo
+B.bar
+
+
+class C
+  def self.foo;
+    class << self
+      class << self
+        def bar; end
+      end
+    end
+  end
+end
```

Comment:
The symbol table output is hard to read and thus brittle to people messing this up. I think you should able to call ` bar` like

```ruby
C.singleton_class.bar
```

Would you mind checking that's how Ruby works, and that there's no static error here?

```
class C
  def self.foo;
    class << self
      class << self
        puts self
        def bar
          'helo'
        end
      end
    end
  end
end

C.foo
p C.singleton_class.bar
```

```
❯ ruby a.rb
#<Class:#<Class:C>>
"helo"
```

Reply:
In fact, I'd almost prefer that we don't even have the symbol table snapshot, and we convert all the things we care about to method calls (with some care taken to guard against `C.singleton_class` somehow becoming `T.untyped`)

Reply:
I've modified the test accordingly to include the `singleton_class.bar` call.